qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -13,17 +13,19 @@
 #  limitations under the License.
 
 from functools import cached_property
-from typing import Dict, Iterable, Sequence, Set, TYPE_CHECKING, Union
+from typing import Any, Dict, Iterable, Sequence, Set, TYPE_CHECKING, Union
 
 import attrs
 import cirq
+import numpy as np
+from cirq._compat import cached_method
```

Comment:
question: what's the difference between this and functools.cache?

Reply:
`functools.cache` caches the first argument as well, which is `self` and therefore you'd end up computing the hash of the class. `cached_method` is smarter and ignores the first argument self